### PR TITLE
UX: Show website path in website name for all domains

### PR DESCRIPTION
**Pull request by mcwumbly**
discussion: https://meta.discourse.org/t/show-full-website-path-on-user-profile-regardless-of-domain/42285?u=mcwumbly

### CODE REVIEW

**Comment by mcwumbly** [comment]
Updated tests:

- Removed redundant tests for domain logic that has been removed
- Added tests to enforce spec regarding truncating query params per discussion on meta

File: app/serializers/user_serializer.rb [hunk 1]
```diff
@@ -145,18 +145,8 @@ def website
...
-      # example.com == example.com
-      website_host + URI(website.to_s).path
-    elsif (website_host.split('.').length == discourse_host.split('.').length) && discourse_host.split('.').length > 2
-      # www.example.com == forum.example.com
-      website_host.split('.')[1..-1].join('.') == discourse_host.split('.')[1..-1].join('.') ? website_host + URI(website.to_s).path : website_host
-    else
-      # example.com == forum.example.com
-      discourse_host.ends_with?("." << website_host) ? website_host + URI(website.to_s).path : website_host
-    end
+    website_host + URI(website.to_s).path
```
Comment (SamSaffron) [inline app/serializers/user_serializer.rb]: no rescue here for URI(website.to_s) this means that bad URI somewhere could just explode the serializer

Comment (mcwumbly) [inline app/serializers/user_serializer.rb]: ok, let me take a look at that

Comment (SamSaffron) [inline app/serializers/user_serializer.rb]: probably worth adding a spec for bad URI

Comment (mcwumbly) [inline app/serializers/user_serializer.rb]: `URI(website.to_s)` is rescued on line 147... I'll add a spec to keep it sane

**Comment by mcwumbly** [comment]
If preferred, method can be rewritten with one call to URI(website.to_s):

```
  def website_name
    uri = URI(website.to_s) rescue nil
    return if uri.nil? || uri.host.nil?
    uri.host + uri.path
  end
```

File: app/serializers/user_serializer.rb (hunk 1 above)
Comment (SamSaffron) [inline app/serializers/user_serializer.rb]: I see, no point double parsing though ... should rewrite it

Comment (mcwumbly) [inline app/serializers/user_serializer.rb]: OK, done.

**Comment by mcwumbly** [comment]
failure looks unrelated to me:
https://travis-ci.org/discourse/discourse/jobs/122143726#L5879

```
Failures:
  1) Scheduler::Manager per host jobs correctly schedules on multiple hosts
     Failure/Error: expect(Testing::PerHostJob.runs).to eq(3)
       
       expected: 3
            got: 2
       
       (compared using ==)
     # ./spec/components/scheduler/manager_spec.rb:98:in `block (3 levels) in <top (required)>'
Finished in 9 minutes 28 seconds (files took 8.51 seconds to load)
5069 examples, 1 failure, 9 pending
Failed examples:
rspec ./spec/components/scheduler/manager_spec.rb:75 # Scheduler::Manager per host jobs correctly schedules on multiple hosts

```

**Comment by SamSaffron** [comment]
There seems to be some logic regarding stripping of first subdomain (I think intended)

If my domain is https://www.samsaffron.com, I think it is reasonable to display `samsaffron.com` so stripping first www. I think is correct. 

Going to merge this in for now, old logic seems very very strange. 

Also need to test what happens to UI when your domain name is:

`https://this-is-very-very-very-very.long.com/with-a-super-long-path`

**Pull request merged**